### ISSUE TITLE
Source code ?

### ISSUE THREAD

**Issue by alkeryn** [X9VoiD/vmulti-bin#6]
Hey, this is nice to provide a signed binary of vmulti, i myself wanted to use it for making some software (added tilt and rotation support) but i don't have signing certificate to sign it.

anyway, would you be open to release the code ?
if not, no worries !

Oh also, if you ever wanna add pen rotation support, that'd be great!

**Comment by Ikxi1** [comment]
The "source" in the releases is also only the binaries :/

**Comment by dtscherer** [comment]
https://github.com/djpnewton/vmulti

**Comment by ikxi3** [comment]
> https://github.com/djpnewton/vmulti

I just found this today too. Didn't even know it was a fork.
Though that repo is like 12-14 years old, has no native pen support with pressure sens & tilt sadly.
At least from what I can see.

**Comment by Ikxi1** [comment]
> https://github.com/djpnewton/vmulti

I just found this today too. Didn't even know it was a fork.
Though that repo is like 12-14 years old, has no native pen support with pressure sens & tilt sadly.
At least from what I can see.

**Comment by alkeryn** [comment]
@dtscherer nope, vmulti-bin is a fork of vmulti that has been modified to add features that the original vmulti does not support.
so the source are not the same.

**Comment by alkeryn** [comment]
@Ikxi1 i already modified the original vmulti to add pressure and tilt, but i can't make proper signed binaries of it because i don't have an EV certificate.

**Comment by dtscherer** [comment]
> @dtscherer nope, vmulti-bin is a fork of vmulti that has been modified to add features that the original vmulti does not support. so the source are not the same.

Good to know!

**Comment by Ikxi1** [comment]
> @Ikxi1 i already modified the original vmulti to add pressure and tilt, but i can't make proper signed binaries of it because i don't have an EV certificate.

OH!? Could you guide me on how you did that?
I've been trying to understand the source code but I'm hitting a wall as I'm not familiar with driver stuff :/

**Comment by alkeryn** [comment]
@Ikxi1 honestly it's been years and i don't remember the whole thing that had to be done, as there is more than one change i did, but 1st you need to change sys/vmulti.h to add descriptors for tilt and pressure, you can get the id's for those descriptors in the windows documentation.

then you'll have to modify the client.c file, specifically the vmulti_update_digi function to account for those change.

the pDigiReport struct will also need to be modified to account for them.

but i think if you take your time you'll figure it out.

otherwise i'll probably opensource it at some point.

the story was that i was making an app to turn tablets into graphic tablets.
but someone beat me to market and i had few weeks left of work.

i then let the project rot because i lost motivation as my potential customer base was taken.

you know what, if you chat with me a bit, i may be willing to give you read access to my vmulti repo.
you can find me with the same username on discord.

**Comment by Ikxi1** [comment]
@alkeryn Thank you!
I'm also trying to turn my android tab into a graphics tab, cause virtual tablet ain't that great and I want more than 60Hz polling rate.
Currently I have everything working besides pressure and tilt with even needing a driver. Could just need some work on latency though.

**Comment by alkeryn** [comment]
@Ikxi1 look up superbrush / superdisplay, that's the competitor that took the market, he honestly did a superb work.
